shared/teams/rename-team/index.js

```diff
@@ -0,0 +1,181 @@
+// @flow
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+
+const splitTeamname = teamname => teamname.split('.')
+
+type Props = {|
+  error?: string,
+  onCancel: () => void,
+  onRename: (newName: string) => void,
+  onSuccess: (newName: string) => void,
+  teamname: string,
+  title: string,
+  waiting: boolean,
+|}
+
+const invalidChars = /[^a-zA-Z0-9_]/
+
+class RenameTeam extends React.Component<Props, {|error: string, newName: string|}> {
+  state = {error: '', newName: ''}
+  _prefix = ''
+  _originalName = ''
+
+  constructor(props: Props) {
+    super(props)
+    const teamNameParts = splitTeamname(this.props.teamname)
+    const newName = teamNameParts.pop()
+    this._prefix = teamNameParts.join('.')
+    this._originalName = newName
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (!this.props.waiting && prevProps.waiting && !this.props.error) {
+      // finished, go back
+      this.props.onSuccess(this._newFullName())
+    }
+  }
+
+  _newFullName = () => [this._prefix, this.state.newName].join('.')
+  _onChangeText = newName => this.setState({newName})
+  _disabled = () => this.state.newName.length < 2
+  _validateTeamname = () => {
```

Comment:
The service errors for these are essentially the same with a bunch of unfriendly codes and stuff. This should catch the most common errors so they can have nice messages.